cli/src/scala/cli/signing/util/BouncycastleSigner.scala

```diff
@@ -33,54 +34,70 @@ final case class BouncycastleSigner(
       val b = content.content()
       f(b, 0, b.length)
     }
+
   def sign(withContent: ((Array[Byte], Int, Int) => Unit) => Unit): Either[String, String] = {
 
     // originally adapted from https://github.com/jordanbaucke/PGP-Sign-and-Encrypt/blob/472d8932df303d6861ec494a3e942ea268eaf25f/src/SignAndEncrypt.java#L144-L199
 
     val encOut = new ByteArrayOutputStream
     val out    = new ArmoredOutputStream(encOut)
 
-    val pgpPrivKey =
+    val maybePgpPrivKey = Try {
       pgpSecretKey.extractPrivateKey(
         new JcePBESecretKeyDecryptorBuilder()
           .setProvider("BC")
           .build(passwordOpt.map(_.value.toCharArray).getOrElse(Array.empty))
       )
+    }.toEither.left.map {
+      case pgpExc: PGPException
+          if pgpExc.getMessage.contains("checksum mismatch at in checksum of") =>
+        passwordOpt match {
+          case Some(_) =>
+            "Failed to decrypt the PGP secret key, make sure the provided password is correct!"
+          case None if pgpSecretKey.getKeyEncryptionAlgorithm != SymmetricKeyAlgorithmTags.NULL =>
+            "Failed to decrypt the PGP secret key, provide a password!"
+          case None => pgpExc.toString
+        }
+      case exc: Throwable =>
+        exc.toString
```

Comment:
Does `exc.toString` also return stack traces? If not, I think we should add stack traces to the string message